src/main/java/mate/academy/spring/dao/RoleDao.java

```diff
@@ -0,0 +1,10 @@
+package mate.academy.spring.dao;
+
+import java.util.Optional;
+import mate.academy.spring.model.Role;
+
+public interface RoleDao {
+    Role add(Role role);
+
+    Optional<Role> getByName(String roleName);
```

Comment:
Common mistakes:
10. Be careful in getByName() method on dao layer! You should pass enum to query (not String), otherwise you won’t be able to get Role by name.
[link](https://mate-academy.github.io/jv-program-common-mistakes/java-spring/security-part-2/jv-spring-security-checklist)